### DISCUSSION TITLE
when I change a file name, or change its location, and the reference isn't updated with it, What should I do

### DISCUSSION THREAD

**Discussion by gengdz** [LazyVim/LazyVim#1227] [Q&A]
What do I need to do to make the reference change

**Comment by folke** [comment]
what do you mean? what is "the reference"?

**Reply by gengdz** [reply]
for example:

There are two files as follows

`src/a.ts`
```ts
// do something
```

`src/b.ts`
```ts
import { xx } from 'src/a.ts'
// do something
```

Then I changed `a.ts` to `a1.ts`, 
I expect `import { xx } from 'src/a.ts` to automatically change path to `import { xx } from 'src/a1.ts` in `b.ts`

**Reply by folke** [reply]
Neovim does not do that automatically. There may be some plugins that support that. Best to google it.

**Reply by ghost** [reply]
Ironic when googling that brings you here

**Reply by abeldekat** [reply]
Hello @davetrust,

I don't understand the meaning of your comment. Could you clarify?

**Reply by mikalcallahan** [reply]
@gengdz youo can have a look at [nvim-lsp-file-operations](https://github.com/antosha417/nvim-lsp-file-operations) which sounds like it will do what you're asking for

**Reply by codingluke** [reply]
Hi there :) I am also very interested in this feature, mostly as I am working a lot on docusaurus pages with cross-markdown links all over the place.

I tired [nvim-lsp-file-operations](https://github.com/antosha417/nvim-lsp-file-operations) together with [marksman](https://github.com/artempyanykh/marksman/issues/153). Unfortunately it is not working. Would be a charme!

Best and thanks for all

**Reply by folke** [reply]
You don't need that file op plugin.
LazyVim supports file renames with references out of the box for neo-tree and mini.files.


Of course, your lsp server also needs to support that.
Works with tsserver for example.

**Reply by codingluke** [reply]
@folke Ah your right, I thought they have implemented it already and it works on emacs but when I read it propperly marksman does not listen to `textDocument/willRename` and `textDocument/didRename`. Goot to know that lazyvim sends them out of the box 🚀
